features/mbedtls/targets/TARGET_CRYPTOCELL310/aes_alt.c

```diff
@@ -707,20 +715,30 @@ int mbedtls_aes_setkey_sw_enc( struct sw_aes_context *ctx, const unsigned char *
 
     switch( keybits )
     {
-        case 128: ctx->nr = 10; break;
-        case 192: ctx->nr = 12; break;
-        case 256: ctx->nr = 14; break;
+        case 128: ctx->keysize_nr = 10; break;
+        case 192: ctx->keysize_nr = 12; break;
+        case 256: ctx->keysize_nr = 14; break;
         default : return( MBEDTLS_ERR_AES_INVALID_KEY_LENGTH );
     }
 
+    if ( NULL == ctx->buf )
+    {
+        ctx->buf = mbedtls_calloc( sizeof(uint32_t) , 68 );
+        if( NULL == ctx->buf )
+        {
+            mbedtls_printf( "Buffer allocation failed\n" );
+            return -1;
```

Comment:
Is there a more descriptive error code here? If there is not one, perhaps you could add one in the `aes_alt.h`? I am not sure what is the policy in this regard though...